controllers/secretgenerator/secretgenerator_controller.go

```diff
@@ -228,11 +228,25 @@ func (r *SecretGeneratorReconciler) createOAuthClient(ctx context.Context, name
 	if err != nil {
 		if apierrs.IsAlreadyExists(err) {
 			secGenLog.Info("OAuth client resource already exists", "name", oauthClient.Name)
-
+			existingOauthClient := &ocv1.OAuthClient{}
```

Comment:
guess this is for:
- check if oauth CR with Name "name" exists:
  -  yes: patch the secretName
  -  no: create it

Reply:
We are recreating it instead of patching it.